projects/packages/publicize/actions.php

```diff
@@ -0,0 +1,19 @@
+<?php
+/**
+ * Action Hooks for Jetpack Publicize module.
+ *
+ * @package automattic/jetpack-publicize
+ */
+
+// If WordPress's plugin API is available already, use it. If not,
+// drop data into `$wp_filter` for `WP_Hook::build_preinitialized_hooks()`.
+if ( function_exists( 'add_action' ) ) {
+	add_action( 'plugins_loaded', array( Automattic\Jetpack\Publicize\Publicize_Assets::class, 'configure' ), 1 );
+} else {
+	global $wp_filter;
```

Comment:
I am not sure ifI have this seen this pattern elsewhere. 


What do you think of this approach in the[ publicize setup class](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/publicize/src/class-publicize-setup.php)?

```
/**
	 * To configure the publicize package, when called via the Config package.
	 */
	public static function configure() {
	         if ( static::$module_active) {
                          		add_action( 'jetpack_feature_publicize_enabled', array( __CLASS__, 'on_jetpack_feature_publicize_enabled' ) );
                  }
                  Automattic\Jetpack\Publicize\Publicize_Assets::class, 'configure' );
                  

	}
```

Reply:
> I am not sure if I have this seen this pattern elsewhere.

As already mentioned in this comment https://github.com/Automattic/jetpack/pull/38606#discussion_r1711089116, we already use that pattern in Jetpack.

> What do you think of this approach in the[ publicize setup class](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/publicize/src/class-publicize-setup.php)?

That class (`Publicize_Setup`) is not even initialized if publicize is not active.

Reply:
> That class (Publicize_Setup) is not even initialized if publicize is not active.

I am proposing to tweak [that](https://github.com/Automattic/jetpack/blob/ce76e6eb0f28f594e9c7273d7329c762a6cf0dda/projects/plugins/jetpack/class.jetpack.php#L913) using the args param the config package can accept. If you see the `class.jetpack.php` file, we are checking if the module is active.  Let's pass whether the module is active down to the publicize setup class. 

Example of using params in the config package: https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/social/src/class-jetpack-social.php#L87  


Thanks for pointing me to the example for an existing pattern in the assets package. I think what I have proposed above is a simpler way to do it.

Reply:
@manzoorwanijk, On second thought, I don't know if it worth's going back and forth on this little detail, if you feel the pattern is good within that, I am happy too. :)

Reply:
That approach may work for only Social but [that is not how it's initialized in Jetpack](https://github.com/Automattic/jetpack/blob/ce76e6eb0f28f594e9c7273d7329c762a6cf0dda/projects/plugins/jetpack/class.jetpack.php#L911-L914).

Reply:
I really do not want to change how the module is initialized as it may affect other unknown areas, especially on WPCOM.

Reply:
Why would it not work for Jetpack? 

```
$config->ensure( 'publicize', [ 'args' => [ 'connected_user' => true, 'module_active' => true] );

````

Reply:
Oh yeah, that way it can work. Let me try that.

Reply:
No no, you don't have to :)

Reply:
No, I think it's worth it.

Reply:
>I really do not want to change how the module is initialized as it may affect other unknown areas, especially on WPCOM.

Fair point! WPCOM is a beast, currently we don't initialize the publicize package fully there.

Reply:
I implemented it with your suggestion and it works but my fear is that it may not work well with older versions of either of the plugins.
For example if we conditionally initialize publicize as you suggested but the user is on an older version of Jetpack or Social, then it won’t work for that plugin.

Thus I am staying with the current approach.

Reply:
Sounds good, appreciate you trying it out :)